sound/soc/sof/loader.c

```diff
@@ -124,8 +124,8 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 			/* They are supported but we don't do anything here */
 			break;
 		default:
-			dev_warn(sdev->dev, "warning: unknown ext header type %d size 0x%x\n",
-				 ext_hdr->type, ext_hdr->hdr.size);
+			dev_dbg(sdev->dev, "unknown ext header type %d size 0x%x\n",
```

Comment:
typo unnessary in commit message

Reply:
Oops, thanks, fixed. The checkpatch codespell doesn't catch much...

Reply:
well it missed one more fraffic haha

Reply:
maybe we need `dev_warn_once()`

Reply:
@lyakh @ktrzcinx but is this a warning? if we have an extendable interface, parser should not be alerted when extensions are used. even the one warning will gather user attention

@ranj063 Oops, heh, I'll address that.

Reply:
> maybe we need `dev_warn_once()`

Or even a `dev_dbg_ocne()` as this logs will show same thing with each PM cycle.

PS: @kv2019i would you think we may only need to parse the manifest in first boot?

Reply:
I see it like this: end user updates FW to have some new functionality (eg. memory scan), kernel update failed/has been forgotten/chosen wrong version during boot, so new feature doesn't work so he need to start debugging/create issue/ask about support. If we keep `if(first_boot) dev_warn(...)` then root cause will be in logs and gathering user attention will be expected and spam after each suspend/resume will be reduced.

Reply:
@ktrzcinx That's indeed useful, but at least for this example, it seems better for sof-logger to detect the debugfs interface is missing and print out an error to user that your kernel is missing memory tracing feature, please upgrade. This is especially true as parsing a manifest header does not mean the kernel has all patches needed for this feature.

@xiulipan That's another option to limit parsing only for first boot. There's a small risk in there, in that we have no guarantees the firmware is not updated on a running system.. it's actually very likely sequence of events when systems are upgraded (there is no forced reboot), so I'd be cautious to build a lot of if-firstboot-then type of logic. Printing warnings like this would maybe qualify (not really an issue if warnings don't get printed until system is fully rebooted).

Even still, let's make sure our warnings and errors do really count. Whenever we emit one, a bug gets reported and it creates work for multiple organizations (integrators, distro maintainers, our customer support and so forth), so even if it's once per boot, it still matters. If the warning can ensure, we can analyze a bug, then it's ok,but let's consider each case on its own merits.

Reply:
@kv2019i Interesting topic, will runtime PM be influenced by firmware change on rootfs?

To me I think the first_boot will be reset with kmod reload and that's when the FW change take effect(we will call require firmware). So we should make ext manifest parser every time we require firmware from rootfs, then we are good with mismatch risk.

I will try to check our FW load sequence to confirm this.

Reply:
@kv2019i I double check the kernel, it seems runtime PM will not output this warning message.

The function `snd_sof_fw_ext_man_parse` is called alongside with `request_firmware`, so this warning will only take effect when boot up or kmod reload.
https://github.com/thesofproject/linux/blob/d0a335dc3ea9cb13846176255fadc78f3bbf9156/sound/soc/sof/loader.c#L687

> As runtime pm is enabled on most platforms and
> snd_sof_fw_parse_ext_data() is called for every DSP boot, this creates
> unnessary kernel dmesg fraffic.

What you said in commit will never happen

Reply:
@xiulipan I now checked the codepaths and I think we are safe with this.

FW:
sof_resume -> snd_sof_load_firmware() -> snd_sof_load_firmware_raw() 

There's a check for "plat_data->fw" and if FW was already loaded, it's not loaded again from rootfs.

Topology is only loaded at sof-audio probe. So if you do a system upgrade on a live system, you need to reload the SOF driver modules to refresh FW and topology, and then all first_boot checks will be done.

Reply:
@kv2019i the commit message mentions IPC data but we're not really talking IPCs here right? Its purely extended manifest headers, isnt it?

Reply:
@ranj063 Yes, I think we have add some ext_manifest both in FW (eg. SOF_IPC_EXT_CC_INFO by IPC for old drviers) and ext_manifest (parse after FW load from rootfs)

@kv2019i I am OK with this as this part will be removed in few versions right? I think it won't break anything as no new feature in driver will relay on the ext data.